### ISSUE TITLE
Clippy treats `/**…*/` doc comments differently from `///…` doc comments

### ISSUE THREAD

**Issue by jbr** [rust-lang/rust-clippy#10277]
### Summary

I had what seemed like a spurious `clippy::missing_errors_doc` that was resolved by changing `/**… */` style doc comment into a `///…` style doc comment. When I did that, additional clippy lints that were previously silent were revealed

### Reproducer

[playground link](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=5e2c1e628428a6fae094123488ba5588) I would have expected the two functions to lint identically, but clippy's missing_errors_doc does not identify the `# Errors` section in one of them. This is not specific to `missing_errors_doc` but seems to apply to lint based on the content of the doc comment. Most of them are false negatives when using `/**…*/`, and as such are silent, but missing_errors_doc is a doc-comment-contents-based false-positive lint.

I would happily contribute a fix with guidance. On initial investigation, maybe [this](https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/doc.rs#L515) is the relevant line?

Thanks!

### Version

```text
rustc 1.69.0-nightly (c8e6a9e8b 2023-01-23)
binary: rustc
commit-hash: c8e6a9e8b6251bbc8276cb78cabe1998deecbed7
commit-date: 2023-01-23
host: x86_64-apple-darwin
release: 1.69.0-nightly
LLVM version: 15.0.7
```

**Comment by Alexendoo** [comment]
They do have different representations, ctrl + f for "# Errors" in the output pane here https://godbolt.org/z/vee1zxbcr

I think you're in the right place yeah, my guess is the issue is here https://github.com/rust-lang/rust-clippy/blob/a2f85deba3483b38059ac18bb317d9dbe7497e76/clippy_lints/src/doc.rs#L472-L474

By playing with rustdoc a bit it looks we should be stripping an amount of whitespace equal to the least indented line in the block comment
